lib/ansible/modules/cloud/amazon/ecs_ecr_facts.py

```diff
@@ -0,0 +1,315 @@
+    returned: for mode=describe_images
+    type: dict of image details, keyed by digest (sha256)
+    example:
+        {
+            "sha256:c31525b2bbd1bfe6a89811cba8a799edab4ab98038fa4a861da56f5b1de31142": {
+                "imageDigest": "sha256:c31525b2bbd1bfe6a89811cba8a799edab4ab98038fa4a861da56f5b1de31142",
+                "imagePushedAt": "2017-03-13T17:32:34-07:00",
+                "imageSizeInBytes": 439173000,
+                "imageTags": [
+                    "latest",
+                    "0.0.3"
+                ],
+                "registryId": "8675309",
+                "repositoryName": "numbers/jenny"
+            }
+        }
+
+
+image_details_by_timestamp:
+    description: image details (creation time, size), sorted (ascending) by creation time
+    returned: for mode=describe_images
+    type: list of image details
+    example:
+        [
+            {
+                "imageDigest": "sha256:c31525b2bbd1bfe6a89811cba8a799edab4ab98038fa4a861da56f5b1de31142",
+                "imagePushedAt": "2017-03-13T17:32:34-07:00",
+                "imageSizeInBytes": 439173000,
+                "imageTags": [
+                    "latest",
+                    "0.0.3"
+                ],
+                "registryId": "8675309",
+                "repositoryName": "numbers/jenny"
+            }
+        ]
+single_result:
+    description: >
+        If only one image was returned (no matter how many were requested), image details for that one image.
+        This is a convenience so Ansible plays don't need to check for number of results when one is assumed.
+    returned: for mode=describe_images
+    type: image details
+    example:
+        {
+            "imageDigest": "sha256:c31525b2bbd1bfe6a89811cba8a799edab4ab98038fa4a861da56f5b1de31142",
+            "imagePushedAt": "2017-03-13T17:32:34-07:00",
+            "imageSizeInBytes": 439173000,
+            "imageTags": [
+                "latest",
+                "0.0.3"
+            ],
+            "registryId": "8675309",
+            "repositoryName": "numbers/jenny"
+        }
+
+'''
+
+import re
+from distutils.version import LooseVersion
+
+from ansible.module_utils.basic import AnsibleModule
+import ansible.module_utils.aws as awsutils
+
+try:
+    import boto3
+
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def list_repositories(ecr):
+    ret = {}
+    res = ecr.describe_repositories()
+    repos = res.get('repositories')
+
+    ret['repo_by_name'] = {v['repositoryName']:v for v in repos}
+    return ret
+
+def list_images(ecr, reponame):
+    ret = {}
+    res = ecr.list_images(repositoryName=reponame)
+
+    # "valid" removes any nonnumbers. This is to remove "latest" but will strip others out.
+    ret['image_by_digest'] = {x['imageDigest']:x for x in res.get('imageIds', [])}
+    ret['valid_image_ids'] = {x['imageTag']:x for x in res.get('imageIds', []) if re.match(r'^[\.0-9]+$', x.get('imageTag', ''))}
```

Comment:
Same for these, also this line is a bit much on the complexity front so would be better as a loop anyhow.